controller/src/beerocks/master/son_master_thread.cpp

```diff
@@ -945,6 +948,72 @@ bool master_thread::handle_cmdu_1905_channel_selection_response(Socket *sd,
 
     } while (cmdu_rx.getNextTlvType() != int(ieee1905_1::eTlvType::TLV_END_OF_MESSAGE));
 
+    return true;
+}
+
+bool master_thread::handle_cmdu_1905_link_metric_responce(Socket *sd, ieee1905_1::CmduMessageRx &cmdu_rx)
+{
+
+    auto mid = cmdu_rx.getMessageId();
+    LOG(INFO) << "Received LINK_METRIC_RESPONSE_MESSAGE, mid=" << std::dec << int(mid);
+
+    // build metric responce message
+    if (!cmdu_tx.create(0, ieee1905_1::eMessageType::LINK_METRIC_RESPONSE_MESSAGE)) {
+        LOG(ERROR) << "cmdu creation of type LINK_METRIC_RESPONSE_MESSAGE, has failed";
+        return false;
+    }
+
+    std::list<son::sTxRxLinkMetricsData> &lCombindMetricData = database.get_combined_metric_data_list(database.get_gw_mac());
+
+
+    // parse all tlvs in cmdu and save ap metric data to database
+    int tlvType;
+    while ((tlvType = cmdu_rx.getNextTlvType()) != int(ieee1905_1::eTlvType::TLV_END_OF_MESSAGE)) {
+
+        if (tlvType < 0) {
+            LOG(ERROR) << "getNextTlvType has failed";
+            return false;
+        }
+
+        if (tlvType == int(ieee1905_1::eTlvType::TLV_TRANSMITTER_LINK_METRIC)) {
+
+            // parse tx_Link_metric_data 
+            auto tx_Link_metric_data = cmdu_rx.addClass<ieee1905_1::tlvTransmitterLinkMetric>();
+            if (!tx_Link_metric_data) {
+                LOG(ERROR) << "addClass ieee1905_1::tx_Link_metric_data has failed";
+                return false;
+            }
+
+            //TODO: save tlvTransmitterLinkMetric parameters to database (lCombindMetricData)
```

Comment:
It will be as simple as db::add_transmitter_link_metric(al_mac_of_reporter, al_mac_of_neighbour, list<sInterfacePairInfo>). Constructing that list is still a bit of code though because the tlvf doesn't do it :-(.